cache-config/testing/docker/trafficserver/run.sh

```diff
@@ -57,31 +83,42 @@ else
   source scl_source enable gcc-toolset-9
 fi
 
-initBuildArea
+cd /root
+# prep build environment
+[ -e rpmbuild ] && rm -rf rpmbuild
+[ ! -e rpmbuild ] || { echo "Failed to clean up rpm build directory 'rpmbuild': $?" >&2; exit 1; }
+mkdir -p rpmbuild/{BUILD,BUILDROOT,RPMS,SPECS,SOURCES,SRPMS} || die "Failed to create build directory '$RPMBUILD': $?"
 
 cd /root/rpmbuild/SOURCES
 # clone the trafficserver repo
-git clone https://github.com/apache/trafficserver.git
+git clone https://github.com/apache/trafficserver.git --branch $ATS_VERSION || die "Failed to fetch the ATS Source"
+cp /traffic_server_jemalloc .
+
+# patch in the astats plugin
+(cp -far /astats_over_http /root/rpmbuild/SOURCES/trafficserver/plugins/astats_over_http
 
-# build trafficserver version 9
+ed /root/rpmbuild/SOURCES/trafficserver/plugins/Makefile.am << ED
```

Comment:
The stock CentOS 7 and CentOS 8 Docker images do not include `ed`.

Reply:
ed was and is installed, see the trafficserver/Dockerfile

Reply:
Although using `ed` works, if `wQ` is changed to `wq` in the edscript, you can use `ex`, which CentOS does include. Because lowercase `wq` works in `ed`, this is not a BC break.

Reply:
changed